pkg/topology/run.go

```diff
@@ -31,30 +32,37 @@ func mergeComponentLookup(ctx *ComponentContext, component *v1.ComponentSpec, sp
 	if err != nil {
 		return nil, errors.Wrapf(err, "component lookup failed: %s", component)
 	}
+
 	if len(results) == 1 {
-		if err := json.Unmarshal([]byte(results[0].(string)), &components); err != nil {
-			return nil, errors.Wrapf(err, "component lookup returned invalid json: %s", component)
-		}
-	} else {
-		// the property returned a list of new properties
-		for _, result := range results {
-			var p pkg.Component
-			data, err := json.Marshal(result)
-			if err != nil {
-				return nil, fmt.Errorf("error marshaling result to json: %w", err)
-			}
-			if err := json.Unmarshal(data, &p); err != nil {
-				return nil, fmt.Errorf("error unmarshaling data from json: %w", err)
+		switch v := results[0].(type) {
```

Comment:
Can't this cause an index out of bounds?

Reply:
There's a length check above it